LagFreeScreenshots/LagFreeScreenshotsMod.cs

```diff
@@ -0,0 +1,526 @@
+                            // flip bit order (endianness)
+                            (output->r, output->g, output->b) = (input->b, input->g, input->r);
+                        }
+                }
+            }
+
+            bitmap.UnlockBits(bitmapData);
+            Marshal.FreeHGlobal(pixelsPair.Item1);
+
+            var format = ourFormat.Value == "auto" ? (hasAlpha ? "png" : "jpeg") : ourFormat.Value;
+            var description = metadata?.ToString();
+
+            // https://docs.microsoft.com/en-us/windows/win32/gdiplus/-gdiplus-constant-property-item-descriptions
+            if (description != null)
+            {
+                // png description is saved as iTXt chunk manually
+                if (format == "jpeg")
+                {
+                    var stringBytesCount = Encoding.Unicode.GetByteCount(description);
+                    var allBytes = new byte[8 + stringBytesCount];
+                    Encoding.ASCII.GetBytes("UNICODE\0", 0, 8, allBytes, 0);
+                    Encoding.Unicode.GetBytes(description, 0, description.Length, allBytes, 8);
+
+                    var pi = (PropertyItem) FormatterServices.GetUninitializedObject(typeof(PropertyItem));
+                    pi.Type = 7; // PropertyTagTypeUndefined
+                    pi.Id = 0x9286; // PropertyTagExifUserComment
+                    pi.Value = allBytes;
+                    pi.Len = pi.Value.Length;
+                    bitmap.SetPropertyItem(pi);
+                }
+            }
+
+            if (format == "jpeg")
+            {
+                var encoder = GetEncoder(ImageFormat.Jpeg);
+                using var parameters = new EncoderParameters(1)
+                {
+                    Param = {[0] = new EncoderParameter(System.Drawing.Imaging.Encoder.Quality, ourJpegPercent.Value)}
+                };
+                filePath = Path.ChangeExtension(filePath, ".jpg");
+                bitmap.Save(filePath, encoder, parameters);
+            }
+            else
+            {
+                filePath = Path.ChangeExtension(filePath, ".png");
+                bitmap.Save(filePath, ImageFormat.Png);
+                if (description != null)
+                {
+                    using var pngStream = new FileStream(filePath, FileMode.Open, FileAccess.ReadWrite);
+                    var originalEndChunkBytes = new byte[12];
+                    pngStream.Position = pngStream.Length - 12;
+                    pngStream.Read(originalEndChunkBytes, 0, 12);
+                    pngStream.Position = pngStream.Length - 12;
+                    var itxtChunk = PngUtils.ProducePngDescriptionTextChunk(description);
+                    pngStream.Write(itxtChunk, 0, itxtChunk.Length);
+                    pngStream.Write(originalEndChunkBytes, 0, originalEndChunkBytes.Length);
+                }
+            }
+
+            await TaskUtilities.YieldToMainThread();
+
+            MelonLogger.Msg($"Image saved to {filePath}");
```

Comment:
Consider using the instance of MelonLogger provided to the mod (same for the other occurrences)

Reply:
The joy of static functions… so we can't :/